### PR TITLE
adding unfreeze at the beginning of the training step

### PR DESCRIPTION
**Pull request by aga0**
to add evaluate memory issues quickly, unfreeze step is added at the beginning of the training (goal: training fails quickly and we can change parameters instead i.e. waiting 2h to see this result)

### CODE REVIEW

**Review by mariaogryczak — COMMENTED** [review]
What is this #full model comment for?

**Comment by aga0** [comment]
> What is this #full model comment for?

ah, you're right, it's not adding any value now - initially I was testing adding extra trainer and this is a leftover I missed, I'll clean it

File: models/model.py [hunk 1]
```diff
@@ -133,6 +133,10 @@ def training_step(self, batch, batch_idx):
         targets = target["target"]
         predictions = self(input)
         loss = self.summary_loss(predictions, targets)
+        if self.global_step > 2:
```
Comment (polquant) [inline models/model.py:136]: If I understand correctly you are unfreezing everything, then immediately you apply freezing schedule. It causes unnecessary overhead.

Comment (mariaogryczak) [reply models/model.py:136]: I think this was actually a goal of this task :D

**Review by mariaogryczak — APPROVED** [review]

File: models/model.py (hunk 1 above)
Comment (polquant) [reply models/model.py:136]: I would add this functionality as an option to the configuration yaml and enter this step only once - we need additional variable if check has been already performed.

Comment (AgaMiko) [reply models/model.py:136]: @polquant "we need additional variable if check has been already performed." 

` self.global_step` calculates a global step, it does not reset every epoch so I don't think there is a need for such a variable. 
https://pytorch-lightning.readthedocs.io/en/latest/common/lightning_module.html#global-step

File: models/model.py [hunk 2]
```diff
@@ -133,11 +133,11 @@ def training_step(self, batch, batch_idx):
         targets = target["target"]
         predictions = self(input)
         loss = self.summary_loss(predictions, targets)
-        if self.global_step>2:
+        if self.global_step > 2:
```
Comment (mplantykow) [inline models/model.py:136]: <

**Review by majsylw — APPROVED** [review]
please resolve current conflicts and merge ;-)

**Pull request merged**